Adds explicit statement of need to review checklist (#862)
openjournals/joss: docs/review_checklist.md

### Documentation

> - **A statement of need:** Do the authors have a Statement of Need that clearly states what problems the software is designed to solve and who the target audience is?

Reviewer: Should it be yet more specific as in "Do the authors have a section entitled Statement of Need..." since that is what exactly is checked for?

> - **A statement of need:** Do the authors have a Statement of Need that clearly states what problems the software is designed to solve and who the target audience is?

Reviewer 2: ```suggestion
- **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
```
